files/image_config/interfaces/interfaces.j2

```diff
@@ -17,34 +17,77 @@ iface lo {{ 'inet' if minigraph_lo_interface['addr'] | ipv4 else 'inet6' }} stat
 #
 {% endfor %}
 {% endblock loopback %}
-#
 {% block mgmt_interface %}
 # The management network interface
 auto eth0
 {% if minigraph_mgmt_interface['addr'] %}
 iface eth0 inet static
     address {{ minigraph_mgmt_interface['addr'] }}
     netmask {{ minigraph_mgmt_interface['mask'] }}
-    ################ management network policy routing rules
-    #### management port up rules"
+    ########## management network policy routing rules
+    # management port up rules
     up ip route add default via {{ minigraph_mgmt_interface['gwaddr'] }} dev eth0 table default
     up ip rule add from {{ minigraph_mgmt_interface['addr'] }}/32 table default
-    #### management port down rules"
+    # management port down rules
     down ip route delete default via {{ minigraph_mgmt_interface['gwaddr'] }} dev eth0 table default
     down ip rule delete from {{ minigraph_mgmt_interface['addr'] }}/32 table default
     {# TODO: COPP policy type rules #}
 {% else %}
 iface eth0 inet dhcp
 {% endif %}
-{% endblock mgmt_interface %}
 #
-{% block front_panel_interface %}
+{% endblock mgmt_interface %}
+{% block front_panel_interfaces %}
 # The switch front panel interfaces
 {% for interface in minigraph_interfaces %}
 auto {{ interface['alias'] }}
+allow-hotplug {{ interface['alias'] }}
 iface {{ interface['alias'] }} {{ 'inet' if interface['addr'] | ipv4 else 'inet6' }} static
     address {{ interface['addr'] }}
     netmask {{ interface['mask'] }}
 #
 {% endfor %}
-{% endblock front_panel_interface %}
+{% for vlan_interface in minigraph_vlan_interfaces|unique_name %}
+{% for interface in vlan_interface['members'] %}
+auto {{ interface }}
+allow-hotplug {{ interface }}
+iface {{ interface }} inet manual
+    pre-up ifconfig {{ interface }} up
+    post-up brctl addif {{ vlan_interface['name'] }} {{ interface }}
+    post-down ifconfig {{ interface }} down
+#
+{% endfor %}
+{% endfor %}
+{% for pc_interface in minigraph_portchannel_interfaces|unique_name %}
+{% for interface in pc_interface['members'] %}
+{% if pc_interface['name'] not in pc_set %}
+auto {{ interface }}
+allow-hotplug {{ interface }}
+iface {{ interface }} inet manual
+    pre-up teamdctl {{ pc_interface['name'] }} port add {{ interface }}
```

Comment:
i think under the teamd interface section, we could add same commands so that once the teamd interface goes up, it will also try to add all its members. there is no harm adding one member twice.